### DISCUSSION TITLE
Multithreading, how it can handle many async requests at the same time?

### DISCUSSION THREAD

**Discussion by GabrielBrotas** [nodejs/help#4974] [Q&A] [answered]
Hi Guys, I understand that Node.js uses a single-thread and an event loop to process requests only processing one at a time (which is non-blocking) and all async functions (I/O operations, File System, DNS resolution,...) goes to another internal thread pool (which has 4 threads by default, right?) to be resolved and then return the response in the callback to the event loop. 
But still, how does that work, lets say 10,000 concurrent requests to an api route that has to wait a response from another api. since it has just 1 thread to dns lookup, does that mean this thread will be blocked? The event loop will process all the requests? the thread pool will have to wait request by request to return the response via callback?
Is there a way to create an elastic thread poll? so we can have 1 thread for each async operation?

**Comment by devsnek** [comment, marked as answer]
duplicate of https://github.com/nodejs/help/issues/3971
